core/jvm/src/main/scala-2.12/reftree/core/HackedCollectionInstances.scala

```diff
@@ -46,84 +46,126 @@ trait HackedCollectionInstances extends CollectionInstances {
           .toField.withName("bitmap")
         RefTree.Ref(trie, Seq(size, bitmap, elems)).rename("HashSet.HashTrieSet")
-      case empty ⇒
+      case empty =>
         RefTree.Ref(empty, Seq.empty).rename("HashSet.EmptyHashSet")
     }
 
   implicit def `HashMap RefTree`[A: ToRefTree, B: ToRefTree]: ToRefTree[HashMap[A, B]] =
     ToRefTree[HashMap[A, B]] {
-      case leaf: HashMap.HashMap1[A, B] ⇒
+      case leaf: HashMap.HashMap1[A, B] =>
         val hash = RefTree.Val.formatted(leaf.privateField[Int]("hash"))(_.toHexString)
           .toField.withName("hash")
         val key = leaf.privateField[A]("key").refTree.toField
         val value = leaf.privateField[A]("value").refTree.toField
         RefTree.Ref(leaf, Seq(hash, key, value)).rename("HashMap.HashMap1")
-      case collision: HashMap.HashMapCollision1[A, B] ⇒
+      case collision: HashMap.HashMapCollision1[A, B] =>
         val hash = RefTree.Val.formatted(collision.privateField[Int]("hash"))(_.toHexString)
           .toField.withName("hash")
         val kvs = collision.privateField[ListMap[A, B]]("kvs").refTree.toField
         RefTree.Ref(collision, Seq(hash, kvs)).rename("HashMap.HashMapCollision1")
-      case trie: HashMap.HashTrieMap[A, B] ⇒
+      case trie: HashMap.HashTrieMap[A, B] =>
         val size = trie.privateField[Int]("size0").refTree.toField.withName("size")
-        val elems = trie.privateField[Array[HashMap[A, B]]]("elems").refTree.toField
-        val bitmap = RefTree.Val.formatted(trie.privateField[Int]("bitmap"))(_.toBinaryString)
-          .toField.withName("bitmap")
+        val elems =
+          trie.packagePrivateField[Array[HashMap[A, B]]](
+            "elems0", "scala.collection.immutable.HashMap$HashTrieMap"
+          ).refTree.toField
+        val bitmap =
+          RefTree.Val.formatted(
+            trie.packagePrivateField[Int](
+              "bitmap0", "scala.collection.immutable.HashMap$HashTrieMap"
+            )
+          )(_.toBinaryString).toField.withName("bitmap")
         RefTree.Ref(trie, Seq(size, bitmap, elems)).rename("HashMap.HashTrieMap")
-      case empty ⇒
+      case empty =>
         RefTree.Ref(empty, Seq.empty).rename("HashMap.EmptyHashMap")
     }
 
   private def redBlackTreeRefTree[A: ToRefTree, B: ToRefTree](
-    tree: RedBlackTree.Tree[A, B],
+    tree: NewRedBlackTree.Tree[A, B],
     includeValue: Boolean
-  ): RefTree = {
-    if (tree == null) RefTree.Null() else {
+  ): RefTree =
+    if (tree != null) {
       val key = tree.key.refTree.toField
       val value = if (includeValue) Seq(tree.value.refTree.toField) else Seq.empty
       val left = redBlackTreeRefTree(tree.left, includeValue).toField
       val right = redBlackTreeRefTree(tree.right, includeValue).toField
+
       RefTree.Ref(tree, Seq(key) ++ value ++ Seq(left, right))
-        .copy(highlight = tree.isInstanceOf[RedBlackTree.RedTree[A, B]])
-    }
-  }
+        .copy(highlight = tree.isInstanceOf[NewRedBlackTree.Tree[A, B]])
```

Comment:
Spotted a regression — the highlight is supposed to be for a red tree, not for all trees. Same in 2.13. Will fix in a separate commit.

Reply:
Hmmm, nice catch. I'm remembering that the whole image was blinking on inserting values within the `gif`. It seems like because of that.

Reply:
I was too eager to finish the migration, so I didn't give it much attention. It's amazing how you spotted this!

Reply:
Well, I rendered a red-black tree and was surprised to see all elements highlighted. I don’t remember much from computer science anymore, but red elements having red children seemed sus :)